smithy-model/src/test/resources/software/amazon/smithy/model/errorfiles/validators/resource-properties/resource-properties.smithy

```diff
@@ -0,0 +1,224 @@
+$version: "2.0"
+
+namespace smithy.example
+
+/// Serves as a somewhat complete and valid example of a
+/// service model with resource defined properties.
+service BookStore {
+    version: "2022-04-01",
+    resources: [
+        Book
+    ]
+    operations: [TagResource, UntagResource]
+}
+
+resource Book {
+    identifiers: {
+        bookId: ISBN
+    }
+    properties: {
+        title: Title
+        publishDate: Date
+        authors: AuthorList
+        price: Price
+        averageReviewScore: Float
+        tagsSillyName: TagList
+    }
+    create: AddBook
+    read: DescribeBook
+    update: UpdateBook
+    delete: DeleteBook
+    operations: [UpdateReviewScore]
+    list: ListBooks
+}
+
+structure BookDescription for Book {
+    $title
+    $publishDate
+    $authors
+    $price
+}
+
+structure Author {
+    givenName: GivenName
+    familyName: FamilyName
+}
+
+list AuthorList {
+    member: Author
+}
+
+operation AddBook {
```

Comment:
nit: inline operation I/O is much nicer

Reply:
I'm going to leave this test cases alone as long as it's valid.